### PR TITLE
Apply Linting Only to Files Updated in git

### LINKED ISSUE #1270: Apply Linting Only to Files Updated in `git`
We need the ability to selectively adjust our linting rules so that they are eventually applied to our entire code base.

1. Linting rules are managed by the `.eslintrc.json` file.
1. They are applied to the entire code base at once.
1. If we change a rule from `warn` to `error` then every file in the code base will be checked and we have frequently seen our tests fail for dozens of files in hundreds of lines of code.

This is not acceptable.

We have been trying to more strictly lint the entire code base manually, but it has been very slow.

## At the user CLI
We need a way to edit `.eslintrc.json` with increasingly stricter linting rules in such a way that the tests are only applied to the changed files with every `git commit` and `git push` command. 

These `git` commands must fail if the linting fails.

## With Every Pull Request

We recently merged a Talawa-API pull request that checks each submitted file for linting compliance. This can be used as an example of how to run the test against individual files.

- https://github.com/PalisadoesFoundation/talawa-admin/pull/1242

You must make sure that this functionality applied for all PRs for this repository whenever the  `.eslintrc.json` file is updated

## Other

1. There may be ways to achieve the CLI goals by updating our existing Git Hooks
1. Your solution should preferably work within coding environments such as VScode
1. Please work with @anwersayeed and @tasneemkoushar on this issue
1. Related issues:
    - https://github.com/PalisadoesFoundation/talawa-api/issues/1548

## Screenshots

When complete, you should be able to convert all these  `.eslintrc.json` file lines containing `warn` to `error` and the tests will only be applied to the changed files in `git`

image

### PR DESCRIPTION
**Pull request by Shubh152**
**What kind of change does this PR introduce?**

feature-request

**Issue Number:**

Fixes #1270

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**
Checks linting rules only for files that are staged for commit, will work with any changes to eslint or prettier configuration.
Uses lint-staged package with existing git hooks

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,12 +74,13 @@
     "test": "cross-env NODE_ENV=test node scripts/test.js --env=./scripts/custom-test-env.js --watchAll --coverage",
     "eject": "react-scripts eject",
     "lint:check": "eslint \"**/*.{ts,tsx}\" --max-warnings=0",
-    "lint:fix": "eslint \"**/*.{ts,tsx}\" --fix",
-    "format:fix": "prettier --write \"**/*.{ts,tsx,json,scss,css}\"",
+    "lint:fix": "eslint --fix",
+    "format:fix": "prettier --write",
     "format:check": "prettier --check \"**/*.{ts,tsx,json,scss,css}\"",
```
Comment (noman2002) [inline package.json:79]: Why these changes are required ?

Comment (Shubh152) [reply package.json:79]: Because lint-staged provides the staged files based on glob patterns specified in .lintstagedrc.json to these commands, So the glob patterns hard-coded in the commands need to be removed.

Image from lint-staged npm page:-
Screenshot from 2023-12-30 19-18-04

Comment (noman2002) [reply package.json:79]: Got it, thank you.

**Comment by noman2002** [comment]
@Shubh152 Please fix the conflicting files. Rest looks good to me.
